packages/react/src/hooks/useWatchKPIToken.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    Fetcher,
+    KPIToken,
+    KPITokenWithData,
+    KPI_TOKEN_ABI,
+    OracleWithData,
+    ORACLE_ABI,
+} from "@carrot-kpi/sdk";
+import { ReadContractConfig } from "@wagmi/core";
+import { useEffect, useState } from "react";
+import { useContractReads, useProvider } from "wagmi";
+import { useIPFSGatewayURL } from "./useIPFSGatewayURL";
+import { usePreferDecentralization } from "./usePreferDecentralization";
+
+export function useWatchKPIToken(
+    kpiTokenOrAddress?: KPIToken | string
+): KPITokenWithData | null {
+    const provider = useProvider();
+    const ipfsGatewayURL = useIPFSGatewayURL();
+    const preferDecentralization = usePreferDecentralization();
+
+    const [kpiToken, setKPIToken] = useState<KPIToken | null>(
+        typeof kpiTokenOrAddress === "string" ? null : kpiTokenOrAddress || null
+    );
+    const [kpiTokenWithData, setKPITokenWithData] =
+        useState<KPITokenWithData | null>(null);
+
+    // in case an address was passed, fetch the kpi token
+    useEffect(() => {
+        if (kpiToken) return;
+        if (typeof kpiTokenOrAddress !== "string") return;
+        let cancelled = false;
+        console.log("IM HERE");
```

Comment:
log

Reply:
Fixed in e0f87e035d0b9532b69c06637f49ba451a551f0d